crate/server/src/config/command_line/db.rs

```diff
@@ -0,0 +1,238 @@
+    fn default() -> Self {
+        Self {
+            database_type: None,
+            database_url: None,
+            sqlite_path: PathBuf::from("./sqlite-data"),
+            clear_database: false,
+            redis_master_password: None,
+            redis_findex_label: None,
+        }
+    }
+}
+
+impl DBConfig {
+    /// Initialize the DB parameters based on the configuration
+    ///
+    /// # Parameters
+    /// - `workspace`: The workspace configuration, used to determine the public and shared paths
+    ///
+    /// # Returns
+    /// - The DB parameters
+    ///
+    /// # Errors
+    /// - If both Postgres and MariaDB/MySQL URL are set
+    /// - If `SQLCipher` is set along with Postgres or MariaDB/MySQL URL
+    pub fn init(
+        &self,
+        workspace: &WorkspaceConfig,
+        use_bootstrap_server: bool,
+    ) -> KResult<Option<DbParams>> {
+        Ok(if let Some(database_type) = &self.database_type {
+            Some(match database_type.as_str() {
+                "postgresql" => {
+                    let url = ensure_url(self.database_url.as_deref(), "KMS_POSTGRES_URL")?;
+                    DbParams::Postgres(url)
+                }
+                "mysql" => {
+                    let url = ensure_url(self.database_url.as_deref(), "KMS_MYSQL_URL")?;
+                    DbParams::Mysql(url)
+                }
+                "sqlite" => {
+                    let path = workspace.finalize_directory(&self.sqlite_path)?;
+                    DbParams::Sqlite(path)
+                }
+                "sqlite-enc" => {
+                    let path = workspace.finalize_directory(&self.sqlite_path)?;
+                    DbParams::SqliteEnc(path)
+                }
+                "redis-findex" => {
+                    let url = ensure_url(self.database_url.as_deref(), "KMS_REDIS_URL")?;
+                    // Check if a Redis master password was provided
+                    let redis_master_password = ensure_value(
+                        self.redis_master_password.as_deref(),
+                        "redis-master-password",
+                        "KMS_REDIS_MASTER_PASSWORD",
+                    )?;
+                    // Generate the symmetric key from the master password
+                    let master_key =
+                        RedisWithFindex::master_key_from_password(&redis_master_password)?;
+                    let redis_findex_label = ensure_value(
+                        self.redis_findex_label.as_deref(),
+                        "redis-findex-label",
+                        "KMS_REDIS_FINDEX_LABEL",
+                    )?;
+                    DbParams::RedisFindex(url, master_key, redis_findex_label.into_bytes())
+                }
+                unknown => kms_bail!("Unknown database type: {unknown}"),
+            })
+        } else if use_bootstrap_server {
+            // That is fine: the bootstrap server will be used to configure the database
+            None
```

Comment:
I do not like this idea too much because the business logic must then be known by the caller (for instance that there is no DB config when you use a bootstrap server); I think the whole logic should be at the same location